packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -439,14 +456,16 @@ export default function () {
           }
 
           // add __esModule declaration if this file has any exports
-          if (hasExports && !strict) {
+          if (hasExports) {
             let buildTemplate = buildExportsModuleDeclaration;
             if (this.opts.loose) buildTemplate = buildLooseExportsModuleDeclaration;
 
-            const declar = buildTemplate();
+            const declar = buildTemplate(t.arrayExpression(Object.keys(exportNames).map(function (name) {
+              return t.stringLiteral(name);
+            })));
             declar._blockHoist = 3;
 
-            topNodes.unshift(declar);
+            topNodes = [].concat(declar).concat(topNodes);
```

Comment:
Ahh, missed that, good call. Okay, hopefully we can change this back if we stop using `__esExports`